tests/sles4sap/publiccloud/mr_test_setup_env.pm

```diff
@@ -0,0 +1,60 @@
+# SUSE's openQA tests
+#
+# Copyright 2022 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+#
+# Summary: Set up public cloud instance testing environment for 'mr_test'
+# Maintainer: QE-SAP <qe-sap@suse.de>
+# Tags: TEAM-6709
+
+use warnings;
+use Mojo::Base 'publiccloud::basetest';
+use publiccloud::ssh_interactive qw(select_host_console prepare_ssh_tunnel);
+use testapi;
+use Mojo::File 'path';
+use publiccloud::utils;
+use Data::Dumper;
+
+sub test_flags {
+    return {
+        fatal => 1,
+        milestone => 0,
+        publiccloud_multi_module => 1
+    };
+}
+
+sub run {
+    my ($self, $run_args) = @_;
+    my $mr_test_tar = 'mr_test-master.tar.gz';
+    my $instance = $run_args->{my_instance};
+
+    # Select console on the host, not the PC instance
+    select_host_console();
+
+    # Preserve args for post_fail_hook
+    $self->{provider} = $run_args->{my_provider};    # Required for cleanup
+    $self->{instance} = $run_args->{my_instance};
+
+    # Download mr_test and extract it to '/root' later
+    if (get_var('PUBLIC_CLOUD_SLES4SAP')) {
+        my $tarball = get_var('MR_TEST_TARBALL', "https://gitlab.suse.de/qa/mr_test/-/archive/master/$mr_test_tar");
+        assert_script_run "curl -sk $tarball -o /root/$mr_test_tar";
+    }
+
+    # Copy the code to instance
+    my $remote = $run_args->{my_instance}->username . '@' . $run_args->{my_instance}->public_ip;
+    $instance->scp("/root/$mr_test_tar", "$remote:/tmp");
+    $instance->ssh_assert_script_run("sudo tar zxf /tmp/$mr_test_tar --strip-components 1 -C /root/");
+    record_info("Copy mr_test code to instance OK");
+
+    # Clear $instance->ssh_opts which ommit the known hosts file and strict host checking by default
+    $instance->ssh_opts("");
+    $instance->network_speed_test();
+
+    # Set ssh-tunnel
+    $testapi::username = "bernhard";
```

Comment:
in cloud we have plenty of cases where username is provider specific . For Azure it is `azure` . https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/publiccloud/azure_client.pm#L18 So unless you really know what you are doing I would not touch username . Every provider perfectly know what needs to be used here

Reply:
I chose "bernhard" as it is a very common/popular user used in openQA;
I tried "cloudadmin" but it reports invalid user: https://openqaworker15.qa.suse.cz/tests/67494#step/mr_test_setup_env/37
So I prefer to keep using this "bernhard".

Reply:
Yes, those commands are running in the jumphost, not in the cloud infra. The command right before that one gives it away: https://openqaworker15.qa.suse.cz/tests/67494#step/mr_test_setup_env/33